### PR TITLE
Task jd/j64 add comments markup on episode page

### PR DESCRIPTION
**Pull request by JuliaDzyuba**

### CODE REVIEW

File: frontend/src/components/episode/components/comment/comment.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { Comment as TComment } from 'common/types/types';
+import { getFormattedDate, getDistanceToDateNow } from 'helpers/date/date';
+import styles from './styles.module.scss';
+
+const DEFAULT_AVATAR = 'https://image.flaticon.com/icons/png/512/848/848043.png';
```
Comment (yevhenprybytok) [inline frontend/src/components/episode/components/comment/comment.tsx]: Put fallback avatar img to assets directory

File: frontend/src/components/episode/components/comment/comment.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  comment: TComment;
+};
+
+const Comment: React.FC<Props> = ({ comment }) => (
+  <li className={styles.wrapper}>
+    <div className={styles.avatarWrapper}>
+      <img src={DEFAULT_AVATAR} width="40" height="40" loading="lazy" alt={`${comment.userId}`}/>
+    </div>
+    <div className={styles.intro}>
+      <p className={styles.userName}>{comment.user.firstName}&nbsp;<span>at</span>&nbsp;{getFormattedDate(new Date(comment.createdAt), 'HH:mm')}</p>
```
Comment (yevhenprybytok) [inline frontend/src/components/episode/components/comment/comment.tsx]: Move date formatting out of the template

File: frontend/src/components/episode/components/comments-list/comments-list.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { Comment as TComment } from 'common/types/types';
+import styles from './styles.module.scss';
+import Comment from '../comment/comment';
```
Comment (yevhenprybytok) [inline frontend/src/components/episode/components/comments-list/comments-list.tsx:3]: Comment should be at least on the same level as comments-list component

**Review by yevhenprybytok — CHANGES_REQUESTED** [review]

**Review by Denidzio — DISMISSED** [review]

File: frontend/src/components/episode/components/comments-list/comments-list.tsx (hunk 3 above)
Comment (JuliaDzyuba) [reply frontend/src/components/episode/components/comments-list/comments-list.tsx:3]: What do you mean? how should it be? Now it look like that
\episode\components
    \comment
        comment.tsx
        styles.module.scss
    \comments-list
        comments-list.tsx
        styles.module.scss

File: backend/src/data/repositories/comment/comment.repository.ts [hunk 4]
```diff
@@ -19,6 +19,10 @@ class Comment {
   public create(payload: CommentCreatePayload): Promise<TComment> {
     return this.#CommentModel.query().insert(payload);
   }
+
+  public getAllCommentsByEpisodeId(id: number): Promise<TComment[]> {
```
Comment (ghost) [inline backend/src/data/repositories/comment/comment.repository.ts]:
```suggestion
  public getAllByEpisodeId(id: number): Promise<TComment[]> {
```

File: backend/src/services/comment/comment.service.ts [hunk 5]
```diff
@@ -19,6 +19,10 @@ class Comment {
   public create(payload: CommentCreatePayload): Promise<TComment> {
     return this.#commentRepository.create(payload);
   }
+
+  public getAllCommentsByEpisodeId(id: number): Promise<TComment[]> {
```
Comment (ghost) [inline backend/src/services/comment/comment.service.ts]:
```suggestion
  public getAllByEpisodeId(id: number): Promise<TComment[]> {
```

File: shared/src/helpers/date/date.ts [hunk 6]
```diff
@@ -1 +1,3 @@
 export { formatISO as getFormattedISODate } from 'date-fns';
+export { format as getFormattedDate } from 'date-fns';
```
Comment (ghost) [inline shared/src/helpers/date/date.ts]: move it to `shared/src/helpers/date/get-formatted-date/get-formatted-date.helper.ts`

```
getFormattedDate(date: Date, formatType: DateFormatType)
```

`DateFormatType` should be enum

File: frontend/src/store/comment/reducer.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
+import { createReducer } from '@reduxjs/toolkit';
```
Comment (ghost) [inline frontend/src/store/comment/reducer.ts]: we don't want reducer for comment

File: frontend/src/store/comment/actions.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+const loadComments = createAsyncThunk<Comment[], undefined, AsyncThunkConfig>
+(ActionType.LOAD_COMMENTS, async (_args, { extra }) => {
+  const { commentApi } = extra;
+  const comments = await commentApi.getAll();
+
+  return comments;
+});
+
+const loadCommentsByEpisodeId = createAsyncThunk<Comment[], number, AsyncThunkConfig>
```
Comment (ghost) [inline frontend/src/store/comment/actions.ts]: move these actions to episode reducer

File: frontend/src/store/actions.ts [hunk 9]
```diff
@@ -4,3 +4,4 @@ export * as configurateEpisode from './configurate-episode/actions';
 export * as homepage from './homepage/actions';
 export * as episode from './episode/actions';
 export * as podcast from './podcast/actions';
+export * as comment from './comment/actions';
```
Comment (ghost) [inline frontend/src/store/actions.ts]:
```suggestion
```

File: frontend/src/components/episode/components/comments-list/comments-list.tsx [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+type Props = {
+  comments: TComment[];
+};
+
+const CommentsList: React.FC<Props> = ({ comments }) => {
```
Comment (ghost) [inline frontend/src/components/episode/components/comments-list/comments-list.tsx:9]: CommentsList component should return only `ul` element
move the logic to one level above

**Review by ghost — CHANGES_REQUESTED** [review]

File: backend/src/services/comment/comment.service.ts [hunk 11]
```diff
@@ -1,23 +1,36 @@
...
 
   public getAll(): Promise<TComment[]> {
     return this.#commentRepository.getAll();
   }
 
-  public create(payload: CommentCreatePayload): Promise<TComment> {
-    return this.#commentRepository.create(payload);
+  public async create(payload: CommentCreatePayload): Promise<TComment> {
+    const user = await this.#userRepository.getById(payload.userId);
+    const newComment = { ...payload, user };
```
Comment (ghost) [inline backend/src/services/comment/comment.service.ts]: you already add `withGraphFetched `, so now you don't need to do it 😉
Model does everything!

File: shared/src/common/enums/date/date-type.enum.ts [hunk 12]
```diff
@@ -0,0 +1,5 @@
+enum DateFormatType {
+  HH_mm = 'HH:mm',
```
Comment (ghost) [inline shared/src/common/enums/date/date-type.enum.ts]: make it more readable, something like
```
HOURSE_MINUTES = 'HH:mm'
```
this is just an example. do your best!

File: shared/src/helpers/date/get-formatted-date/get-formatted-date.helper.ts [hunk 13]
```diff
@@ -0,0 +1,3 @@
+import { format as getFormattedDate } from 'date-fns';
+
+export { getFormattedDate };
```
Comment (ghost) [inline shared/src/helpers/date/get-formatted-date/get-formatted-date.helper.ts]:
```suggestion
import { format } from 'date-fns';

const getFormattedDate = (date: Date, formatType: DateFormatType) => {
  return format(date, formatType)
};

export { getFormattedDate };
```

File: frontend/src/store/comment/actions.ts (hunk 8 above)
Comment (ghost) [reply frontend/src/store/comment/actions.ts]: don't fixed

File: frontend/src/store/episode/reducer.ts [hunk 14]
```diff
@@ -1,16 +1,19 @@
 import { createReducer } from '@reduxjs/toolkit';
 import { DataStatus } from 'common/enums/enums';
-import { Episode } from 'common/types/types';
+import { Episode, Comment } from 'common/types/types';
 import { loadEpisode } from './actions';
+import { loadCommentsByEpisodeId } from '../comment/actions';
```
Comment (ghost) [inline frontend/src/store/episode/reducer.ts]:
```suggestion
import { loadEpisode, loadCommentsByEpisodeId } from './actions';
```

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**